### PR TITLE
HV-1095 - Provide module ZIP file for easily upgrading Hibernate Validator in a WildFly instance

### PR DESCRIPTION
**Pull request by marko-bekhta**
- https://hibernate.atlassian.net/browse/HV-1095

+added new maven module for creating a module zip file for WildFly
+configured assembly plugin in new module to create a zip with validator and validator-cdi libs.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Nice one! 

Could you also update the existing integration test module so that it uses this new module ZIP?Currently we just unzip WildFly, replace the existing JARs within the ones from our own build and that's it. Instead, we should unzip WildFly, unzip our new module ZIP and then let the test depend on the specific "slot" of the new modules (similar to what's done in the ORM module integration test for instance).

This also will require to disable the implicitly added HV module dependencies (see [here](https://docs.jboss.org/author/display/WFLY10/Class+Loading+in+WildFly) and [here](https://docs.jboss.org/author/display/WFLY8/Implicit+module+dependencies+for+deployments), so that users's don't have the HV coming with WF and the one from the module ZIP added to their deployment.

File: modules/src/main/modules/org/hibernate/validator/module.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+~ Lesser General Public License for more details.
+~
+~ You should have received a copy of the GNU Lesser General Public
+~ License along with this software; if not, write to the Free
+~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<module xmlns="urn:jboss:module:1.3" name="org.hibernate.validator.cdi">
```
Comment (gunnarmorling) [inline modules/src/main/modules/org/hibernate/validator/module.xml]: name=org.hibernate.validator

File: modules/src/main/modules/org/hibernate/validator/module.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
```
Comment (gunnarmorling) [inline modules/src/main/modules/org/hibernate/validator/module.xml]: Could you add the header we use in other XML files instead.

File: modules/pom.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>hibernate-validator-modules</artifactId>
+
+    <name>Hibernate Validator WildFly Module</name>
+    <description>Hibernate Validator modules creation to use in WildFly</description>
+
+    <properties>
+        <module.xml.basedir>${basedir}/src/main/modules</module.xml.basedir>
+        <hibernate.validator.module.slot>${project.version}</hibernate.validator.module.slot>
```
Comment (gunnarmorling) [inline modules/pom.xml]: That looks good. For ORM and HSEARCH we are using the alias mechanism in addition, allowing to use "6" or "6.1" when referencing the module, making it a bit simpler for the user. Would be nice to have the same here (you can find examples for how it's done in these other projects).

Comment (marko-bekhta) [inline modules/pom.xml]: Thanks for the comments. I'll make the changes today.

**Comment by gunnarmorling** [comment]
Going to close this one, as #523 is based on top of it. Thanks, @marko-bekhta!

**Comment by marko-bekhta** [comment]
great !

**Pull request closed without merging**